### PR TITLE
fix(gradient-block): Position Uniqueness

### PR DESCRIPTION
**Pull request by ragi96**
[Task](https://app.clickup.com/t/8677uqnq0)

UX Review finding which was caused problems if multiple color had the same position.

### CODE REVIEW

File: packages/gradient-block/src/components/ColorFlyout.tsx [hunk 1]
```diff
@@ -58,7 +58,11 @@ export const ColorFlyout = ({
 
     const setValidColorPosition = (value: string) => {
         const valueAsNumber = parseInt(value);
-        if (valueAsNumber >= 0 && valueAsNumber <= 100) {
+        if (
+            valueAsNumber >= 0 &&
+            valueAsNumber <= 100 &&
+            gradientColors.every((item) => item.position !== valueAsNumber)
```
Comment (celine-s) [inline packages/gradient-block/src/components/ColorFlyout.tsx:64]: Maybe we could just add the color to the outermost part. E.g. if the valueAsNumber is 103 => add it to 100 and if it is smaller than 0 add it to 0. Atm I can still add a color to higher than 103, but it doesn't place it: https://www.loom.com/share/2d50c89f1b1f4c08bf2bb8be14a6bc04

Comment (ragi96) [reply packages/gradient-block/src/components/ColorFlyout.tsx:64]: How can you add at position 103? I can't manage to do that

File: packages/gradient-block/src/components/ColorFlyout.tsx [hunk 2]
```diff
@@ -137,7 +141,7 @@ export const ColorFlyout = ({
                     <FormControl
                         helper={
                             colorPositionValidation === Validation.Error
-                                ? { text: 'Add color stops from 0-100.' }
+                                ? { text: 'Add color stops from 0-100 and it needs to be unique.' }
```
Comment (fulopdaniel) [inline packages/gradient-block/src/components/ColorFlyout.tsx:144]:
```suggestion
                                ? { text: 'Add unique color stops from 0-100.' }
```

Comment (fulopdaniel) [reply packages/gradient-block/src/components/ColorFlyout.tsx:144]: Just feels a bit nicer to me but I am fine with both

**Review by fulopdaniel — APPROVED** [review]

File: packages/gradient-block/src/components/SquareBadgesRow.tsx [hunk 3]
```diff
@@ -23,7 +23,9 @@ export const SquareBadgesRow = ({ blockWidth, gradientColors, gradientOrientatio
     };
 
     useEffect(() => {
-        prepareGradients();
+        if (gradientColors.length > 0) {
```
Comment (celine-s) [inline packages/gradient-block/src/components/SquareBadgesRow.tsx:26]: maybe we could display a info text like `Please note that the gradient block requires a minimum of two colors to be selected in order to function properly.` Wdyt?

**Review by celine-s — APPROVED** [review]
just one idea how to solve the UX when there is no or only one color selected. Anyway now it works good - thanks!

File: packages/gradient-block/src/components/SquareBadgesRow.tsx (hunk 3 above)
Comment (ragi96) [reply packages/gradient-block/src/components/SquareBadgesRow.tsx:26]: Yeah i think that would be great, any idea where to put it?

**Review by SamuelAlev — CHANGES_REQUESTED** [review]
Please add tests for your fixes to prevent regression

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**